Sources/Utils/PlaylistReader.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  PlaylistReader.swift
+//  spatialgen-whitelabel
+//
+//  Created by Zachary Handshoe on 8/28/24.
+//
+
+import Foundation
+import CoreGraphics
+
+func getAvailableResolutions(from url: URL, completion: @escaping (Result<[ResolutionOption], Error>) -> Void) {
+    URLSession.shared.dataTask(with: url) { data, response, error in
+        if let error = error {
+            completion(.failure(error))
+            return
+        }
+        
+        guard let data = data, let content = String(data: data, encoding: .utf8) else {
+            completion(.failure(NSError(domain: "ParsingError", code: 0, userInfo: [NSLocalizedDescriptionKey: "Failed to parse m3u8 content"])))
```

Comment:
this error could be also appropriate for `parseResolutions()` returning an empty array

Reply:
Pushed an update to add an error for an empty array.